app/controllers/application_controller.rb

```diff
@@ -14,17 +14,7 @@ def mobile_filter_header
   end
 
    def set_locale
-     I18n.locale = extract_locale_from_accept_language_header
+     I18n.locale = http_accept_language.compatible_language_from(I18n.available_locales)
    end
```

Comment:
Consider using `http_accept_language.`**`language_region_compatible_from`**`(I18n.available_locales)`?

According to [this README](https://github.com/iain/http_accept_language/blob/master/README.md#available-methods):

> **language_region_compatible_from(languages)**: Returns the first of the user preferred languages that is also found in available languages. Finds best fit by matching on primary language first and secondarily on region. If no matching region is found, return the first language in the group matching that primary language.
vs:
> **compatible_language_from(languages)**: Returns the first of the user_preferred_languages that is compatible with the available locales. Ignores region.